atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/metrics/LastSweptTsUpdateScheduler.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * (c) Copyright 2022 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.sweep.metrics;
+
+import com.palantir.common.concurrent.NamedThreadFactory;
+import com.palantir.common.concurrent.PTExecutors;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import java.io.Closeable;
+import java.time.Duration;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public final class LastSweptTsUpdateScheduler implements Closeable {
+    private static final SafeLogger log = SafeLoggerFactory.get(LastSweptTsUpdateScheduler.class);
+    private static final long DELAY = Duration.ofSeconds(30).toMillis();
+    private final ScheduledExecutorService executorService;
+    private final Runnable task;
+
+    private LastSweptTsUpdateScheduler(ScheduledExecutorService executorService, Runnable task) {
+        this.executorService = executorService;
+        this.task = task;
+    }
+
+    public static LastSweptTsUpdateScheduler createStarted(Runnable task) {
+        ScheduledExecutorService executorService = PTExecutors.newSingleThreadScheduledExecutor(
+                new NamedThreadFactory("Last Swept Timestamp Metric Update", true));
+        LastSweptTsUpdateScheduler scheduler = new LastSweptTsUpdateScheduler(executorService, task);
+        scheduler.start();
+        return scheduler;
+    }
+
+    private void runOnce() {
+        try {
+            task.run();
+        } catch (Exception exception) {
+            log.info("Last Swept Timestamp Scheduler task failed ", exception);
```

Comment:
How would you detect if this failed in prod? (Note that this is one of the PR template questions!)

Open questions to make you think:
Will you see the info log? Why / why not?
Do you want metrics for this? 
Do we need a monitor? (what happens if this task doesn't run?)

Ping me if you want some help on how to think through this!